prow/git/git.go

```diff
@@ -95,17 +98,21 @@ func (c *Client) SetRemote(remote string) {
 
 // SetCredentials sets credentials in the client to be used for pushing to
 // or pulling from remote repositories.
-func (c *Client) SetCredentials(user, pass string) {
+func (c *Client) SetCredentials(user string, tokenGenerator func() []byte) {
 	c.credLock.Lock()
 	defer c.credLock.Unlock()
 	c.user = user
-	c.pass = pass
+	c.tokenGenerator = tokenGenerator
 }
 
 func (c *Client) getCredentials() (string, string) {
 	c.credLock.RLock()
 	defer c.credLock.RUnlock()
-	return c.user, c.pass
+	if c.tokenGenerator != nil {
```

Comment:
this should be impossible with the constructor

Reply:
Seems outstanding